onnxruntime/test/providers/cpu/tensor/pad_test.cc

```diff
@@ -743,5 +769,15 @@ TYPED_TEST(PadOpTest, Pad_Reflect_DimWithZeroInput) {
                                   "Cannot use 'reflect' mode to pad dimension with a value of 0. Input shape:{0,2,1}");
 }
 
+TEST(PadOpTest, BoolType) {
+  OpTester test("Pad", 13);
+  test.AddAttribute("mode", "constant");
+  test.AddInput<bool>("data", {3, 2}, {true, false, true, false, true, false});
+  test.AddInput<int64_t>("pads", {4}, {0, 2, 0, 0});
```

Comment:
nit: easier to read if this is 3 rows of 2 values and the expected output is 3 rows of 4 values.